### DISCUSSION TITLE
useDialogPluginComponent

### DISCUSSION THREAD

**Discussion by muscleman** [quasarframework/quasar#10076] [General - Components / Directives / etc]
hi,

this is vue3 and typescript

i have a parent component.

from a menu option1 i would like to open an imported component called c1 as a dialog
from a menu option2 i would like to open an imported component called c2 as a dialog
from a menu option3 i would like to open an imported component called c3 as a dialog

so i followed the full example [here](https://quasar.dev/vue-composables/use-dialog-plugin-component#introduction) and placed inside a separate file for c1

how do i instruct c1, c2 or c3 to open there sibling dialogs? from the option1, 2 or 3 click event

main.vue

    <template>
        <q-btn-dropdown>
        ....
        <q-item clickable v-close-popup @click="openC1"
        <q-item clickable v-close-popup @click="openC2"
        <q-item clickable v-close-popup @click="openC3"
        </q-btn-dropdown>
        <c1></c1>
        <c2></c2>
        <c3></c3>
    </template>
    <script>
    import c1 from './c1.vue'
    import c2 from './c2.vue'
    import c3 from './c3.vue'
    export default defineComponent ({
        name: 'main',
        setup() {
            function openC1() { somehow open c1}
            function openC2() { somehow open c2}
            function openC3() { somehow open c3}
            return {
                openC1,
                openC2,
                openC3,
            }
        },
        components: {
            c1,
            c2,
            c3
        }
    })
    </script>

c1.vue
```
<template>
    <q-dialog ref="dialogRef" @hide="onDialogHide">
        .......stuff
    </q-dialog>
</template>
<script>
import { useDialogPluginComponent } from 'quasar'
import { defineComponent, ref } from 'vue'
export default defineComponent ({
    name: 'c1',
    emits: [
        ...useDialogPluginComponent.emits
    ],
    setup() {
        const { dialogRef, onDialogHide, onDialogOK, onDialogCancel  } = useDialogPluginComponent()
        return {
            dialogRef,
            onDialogHide,
            onOKClick,
            onCancelClick: onDialogCancel
            }
        }
    })
</script>
```

c2.vue
```
<template>
    <q-dialog ref="dialogRef" @hide="onDialogHide">
        .......stuff
    </q-dialog>
</template>
<script>
import { useDialogPluginComponent } from 'quasar'
import { defineComponent, ref } from 'vue'
export default defineComponent ({
    name: 'c2',
    emits: [
        ...useDialogPluginComponent.emits
    ],
    setup() {
        const { dialogRef, onDialogHide, onDialogOK, onDialogCancel  } = useDialogPluginComponent()
        return {
            dialogRef,
            onDialogHide,
            onOKClick,
            onCancelClick: onDialogCancel
            }
        }
    })
</script>
```

c3.vue
```
<template>
    <q-dialog ref="dialogRef" @hide="onDialogHide">
        .......stuff
    </q-dialog>
</template>
<script>
import { useDialogPluginComponent } from 'quasar'
import { defineComponent, ref } from 'vue'
export default defineComponent ({
    name: 'c3',
    emits: [
        ...useDialogPluginComponent.emits
    ],
    setup() {
        const { dialogRef, onDialogHide, onDialogOK, onDialogCancel  } = useDialogPluginComponent()
        return {
            dialogRef,
            onDialogHide,
            onOKClick,
            onCancelClick: onDialogCancel
            }
        }
    })
</script>
```

**Comment by metalsadman** [comment]
@muscleman you aren't using them as a custom component for Dialog plugin. check the docs https://quasar.dev/quasar-plugins/dialog#invoking-custom-component
where `import CustomComponent from '..path.to.component..'` = your c1...c3
Take note of the tip 
![image](https://user-images.githubusercontent.com/31038838/125553991-9a47decc-a859-463e-bb64-0c7421fd5aee.png)
from https://quasar.dev/vue-composables/use-dialog-plugin-component#introduction

From the way you are implementing it (normal QDialog component), you can just use refs on your c1/c2/c3 components.
ie. 
```html
<template>
    <q-btn-dropdown>
    ....
    <q-item clickable v-close-popup @click="openC1"
    <q-item clickable v-close-popup @click="openC2"
    <q-item clickable v-close-popup @click="openC3"
    </q-btn-dropdown>
    <c1 ref="dialog1"></c1>
    <c2 ref="dialog2"></c2>
    <c3 ref="dialog3"></c3>
</template>
```
```js
...
setup() {
        // register refs
        const dialog1 = ref(null),
          dialog2 = ref(null),
          dialog3 = ref(null)
        function openC1() { dialog1.value.dialogRef.value.show() }
        function openC2() { dialog2.value.dialogRef.value.show() }
        function openC3() { dialog3.value.dialogRef.value.show() }
        return {
            openC1,
            openC2,
            openC3,
            dialog1,
            dialog2,
            dialog3
        }
    },
...
```

**Comment by muscleman** [comment]
Still fail to see how to apply this to my requirements

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Reply by metalsadman** [reply]
better post a codepen so we can see, i'm pretty sure it works.

**Reply by muscleman** [reply]
i'll try get something in codepen next week. thanks

**Reply by muscleman** [reply]
hmm, might need some help with the codepen, not sure how to import another component. lol.

https://codepen.io/musclesonvacation/pen/jOmBxxP?editors=1111
